pkg/pillar/zedUpload/ociutil/oci.go

```diff
@@ -100,10 +100,7 @@ func PullBlob(registry, repo, hash, localFile, username, apiKey string, maxsize
 
 	// The OCI distribution spec only uses /blobs/ endpoint for layers or config, not index or manifest.
 	// I have no idea why you cannot get a manifest or index from the /blobs endpoint, but so be it.
-	image := repo
-	if hash != "" {
-		image = fmt.Sprintf("%s@%s", repo, hash)
-	}
+	image := fmt.Sprintf("%s/%s", registry, repo)
```

Comment:
Will this work? We pass the actual hash in, and here we ignore it, leading us to get nothing for L119:

```go
		d, ok := ref.(name.Digest)
```

Reply:
As per our current flow, `repo` already has an image hash, so we don't need to add an image hash.

Reply:
I thought that sometimes it does, and sometimes it doesn't? Yes, I know, I wrote much of that `zedUpload/ociutil/`, but doesn't mean I remember it. Is it not possible to sometimes have no hash? I don't mean that we never actually do that, but that it is possible to hit that code flow?

Reply:
@adarsh-zededa if that is the case, then why are you changing image on line 105? I'm confused.

Reply:
@nordmark I actually updated the PR, but for some reason it's not showing up here, I'm changing the `image` at line 103.
This is the last commit: https://github.com/lf-edge/eve/pull/1242/commits/f03e8b07e879ab6ab07740a913d66df6f5a5e949

Reply:
And you change the image variable on line 105 (old code) which doesn't seem to match your explanation

Reply:
I handled image at line 105. Thanks @eriknordmark!